krabs/krabs/size_provider.hpp

```diff
@@ -70,6 +70,15 @@ namespace krabs {
             // length is a union that may refer to another field for a length
             // value. In that case, defer to TDH for the value otherwise
             // use the length value directly.
+
+            // Some providers set length == 8 for 32bit pointer values. But
+            // only put 4 bytes into the buffer. Force correct size for pointers.
+            if (propertyInfo.nonStructType.InType == TDH_INTYPE_POINTER &&
```

Comment:
I want to understand this a little more. It seems like this could be very error prone depending on whether the logging process in 32 or 64. Did you find documentation about this? If not, we should look into this harder.

Reply:
https://github.com/thnetii/windows-sdk-headers/blob/master/um/tdh.h#L164 seems to give a hint on this. E.g. it suggests that the header should be used to determine size. I can try to dig out the exact events I was observing this with if you want to investigate further. Just let me know.

Reply:
![krabs_case](https://user-images.githubusercontent.com/40087/76347534-7bbf1100-630f-11ea-9786-55cd4aa06c65.png)

So example in the screenshot. UserData (0x00000125a7e601c8) is shown in the memory view. And in the memory one can see that the pointer is 4 bytes. At byte 5 there is a string starting. But propertyInfo.length is 8. So this would cause krabs to give out corrupted data.

This is from provider {43D1A55C-76D6-4F7E-995C-64C711E5CAFE} and event id is 1179 . This is produced by using Edge on Win 10 1903.

Reply:
This also looks related: https://docs.microsoft.com/en-us/windows/win32/api/tdh/nf-tdh-tdhformatproperty

`PointerSize`: The size of a pointer value in the event data. To get the size, access the `EVENT_RECORD.EventHeader.Flags` member. The pointer size is 4 bytes if the `EVENT_HEADER_FLAG_32_BIT_HEADER` flag is set; otherwise, it is 8 bytes if the `EVENT_HEADER_FLAG_64_BIT_HEADER` flag is set.

Reply:
If that's the case then instead of this condition, maybe we should have something along these lines:
```
if (is_pointer) { return record.EventHeader.Flags & EVENT_HEADER_FLAG_32_BIT_HEADER ? 4 : 8; }
```

Reply:
Agreed, that is better for readability.

Reply:
Updated the PR accordingly